manifests/plugin/python.pp

```diff
@@ -20,26 +15,39 @@
   }
   # End deprecation
 
-  file {
-    "${name}.load":
-      ensure  => $ensure,
-      path    => "${conf_dir}/${order}-${name}.conf",
-      owner   => 'root',
-      group   => $collectd::params::root_group,
-      mode    => '0640',
-      content => template('collectd/python.conf.erb'),
-      notify  => Service['collectd'],
+  $ensure_modulepath = $ensure ? {
+      'absent' => $ensure,
+      default  => 'directory',
   }
 
-  file {
-    "${name}.script":
-      ensure  => $ensure,
-      path    => "${modulepath}/${module}.py",
-      owner   => 'root',
+  file { $modulepath :
+      ensure  => $ensure_modulepath,
+      mode    => '0750',
+      owner   => root,
       group   => $collectd::params::root_group,
-      mode    => '0640',
-      source  => $script_source,
-      require => File["${name}.load"],
-      notify  => Service['collectd'],
+      require => Package[$collectd::params::package]
+  }
+
+  collectd::plugin { 'python' :
+      ensure   => $ensure,
+      interval => $interval,
```

Comment:
Except that you missed this part: "If a plugin provides own support for specifying an interval, that setting will take precedence.". In the case of the Python plugin, there is no support for custom intervals so the parameter is ineffective. Here is what I get on one of my boxes:

    python plugin: Ignoring unknown config key "Interval"